engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingminmax/ComparableRollingMinMaxOperator.java

```diff
@@ -0,0 +1,148 @@
+                    if (a == null) {
+                        return b;
+                    } else if (b == null) {
+                        return a;
+                    }
+                    return a.compareTo(b) > 0 ? a : b;
+                });
+            } else {
+                aggMinMax = new AggregatingObjectRingBuffer<>(BUFFER_INITIAL_CAPACITY, null, (a, b) -> {
+                    if (a == null) {
+                        return b;
+                    } else if (b == null) {
+                        return a;
+                    }
+                    return a.compareTo(b) < 0 ? a : b;
+                });
+            }
+            evaluationNeeded = false;
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            aggMinMax = null;
+        }
+
+
+        @Override
+        public void setValuesChunk(@NotNull final Chunk<? extends Values> valuesChunk) {
+            objectInfluencerValuesChunk = valuesChunk.asObjectChunk();
+        }
+
+        @Override
+        public void push(int pos, int count) {
+            aggMinMax.ensureRemaining(count);
+
+            for (int ii = 0; ii < count; ii++) {
+                T val = objectInfluencerValuesChunk.get(pos + ii);
+                aggMinMax.addUnsafe(val);
+
+                if (val == null) {
+                    nullCount++;
+                } else {
+                    if (curVal == null) {
+                        curVal = val;
+                    } else if (isMax && curVal.compareTo(val) < 0) {
+                        curVal = val;
+                        evaluationNeeded = false;
+                    } else if (!isMax && curVal.compareTo(val) > 0) {
+                        curVal = val;
+                        evaluationNeeded = false;
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void pop(int count) {
+            Assert.geq(aggMinMax.size(), "shortWindowValues.size()", count);
+
+            for (int ii = 0; ii < count; ii++) {
+                T val = aggMinMax.removeUnsafe();
+
+                if (val == null) {
+                    nullCount--;
+                } else {
+                    // Only revaluate if we pop something equal to our current value. Otherwise we have perfect
+                    // confidence that the min/max is still in the window.
+                    if (curVal != null && curVal.compareTo(val) == 0) {
+                        evaluationNeeded = true;
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void writeToOutputChunk(int outIdx) {
+            if (aggMinMax.size() == nullCount) {
+                outputValues.set(outIdx, null);
+            } else {
+                if (evaluationNeeded) {
+                    curVal = aggMinMax.evaluate();
+                }
+                outputValues.set(outIdx, curVal);
+                evaluationNeeded = false;
```

Comment:
should this `evaluationNeeded` pop out of the else? (as in, when all elements are null, can we considered that evaluated?)

If you do set `curVal` to null, then you can pull the `outputValues.set` outside of the else block too... looking something like this:

```
if (aggMinMax.size() == nullCount) {
    curVal = null;
} else if (evaluationNeeded) {
    curVal = aggMinMax.evaluate();
}
outputValues.set(outIdx, curVal);
evaluationNeeded = false;
```